.github/workflows/testall.yaml

```diff
@@ -0,0 +1,153 @@
+name: main
+
+on:
+  workflow_call:
+    inputs:
+      build-artifact-name:
+        required: true
+        type: string
+      build-artifact-path:
+        required: true
+        type: string
+      self-build:
+        required: false
+        type: number
+      pyodide-versions:
+        required: false
+        type: string
+        default: "*"
+      runners:
+        required: false
+        type: string
+        default: '[selenium,playwright]'
+      browsers:
+        required: false
+        type: string
+        default: "*"
+      os:
+        required: false
+        type: string
+        default: '[ubuntu-latest,macos-latest]'
+permissions:
+  contents: read
+jobs:
+  make_test_matrix:
+    outputs:
+      matrix: ${{ steps.filter.outputs.matrix }}
+      pyodide_versions: ${{ steps.filter.outputs.pyodide_versions }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Write py file
+        run: |
+          cat << 'EOF' > filtermatrix.py
```

Comment:
As @rth mentioned, can we actually have this as a standalone `.py` file?

Reply:
done